src/CometWrapper.sol

```diff
@@ -331,16 +331,29 @@ contract CometWrapper is ERC4626, CometHelpers {
     /// @param assets The amount of assets to deposit
     /// @return The total amount of shares that would be minted by the deposit
     function previewDeposit(uint256 assets) public view override returns (uint256) {
-        return convertToShares(assets);
+        // Calculate shares to mint by calculating the new principal amount
+        uint64 baseSupplyIndex_ = accruedSupplyIndex();
+        uint256 currentPrincipal = totalSupply;
```

Comment:
what is the purpose of this assignment? solmateErc20 total suplpy is uint256, is it not?

Reply:
It's really only used for readability, since we're comparing it against the `newPrincipal`. You're correct that it's already an uint256.